.github/workflows/psalm.yml

```diff
@@ -0,0 +1,97 @@
+
+on:
+  pull_request: ~
+
+jobs:
+  psalm:
+    name: Psalm
+    runs-on: Ubuntu-20.04
+
+    steps:
+      - name: Set up PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: '8.0'
+          extensions: "json,memcached,mongodb,redis,xsl,ldap,dom"
+          ini-values: "memory_limit=-1"
+          coverage: none
+
+      - name: Checkout base
+        uses: actions/checkout@v2
+        with:
+          path: pr
+
+      - name: Checkout PR
+        uses: actions/checkout@v2
+        with:
+          ref: git merge-base --fork-point ${{ github.base_ref }} ${{ github.ref }}
+          path: base
+
+      - name: Configure composer
+        run: |
+            cd base
+            COMPOSER_HOME="$(composer config home)"
+            ([ -d "$COMPOSER_HOME" ] || mkdir "$COMPOSER_HOME") && cp .github/composer-config.json "$COMPOSER_HOME/config.json"
+            echo "COMPOSER_ROOT_VERSION=$(grep -m1 SYMFONY_VERSION .travis.yml | grep -o '[0-9.x]*').x-dev" >> $GITHUB_ENV
+
+      - name: Determine composer cache directory
+        id: composer-cache
+        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+
+      - name: Cache composer dependencies
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.composer-cache.outputs.dir }}
+          key: composer-${{ github.sha }}
+          restore-keys: composer-
+
+      - name: Install Psalm
+        run: |
+          composer require psalm/phar
+          cp ./vendor/bin/psalm.phar base/psalm.phar
+          cp ./vendor/bin/psalm.phar pr/psalm.phar
+
+      - name: Install dependencies for base
+        run: |
+            cd base
+            echo "::group::modify composer.json"
+            sed -i -re 's/"replace": \{/"replace": \{"symfony\/phpunit-bridge": "self.version",/' composer.json
+            composer require --no-update phpunit/phpunit php-http/discovery
+            echo "::endgroup::"
+            echo "::group::composer update"
+            composer update --no-progress --ansi
+            echo "::endgroup::"
+
+      - name: Generate Psalm baseline
+        run: |
+          cd base
+          ./psalm.phar --set-baseline=psalm.baseline.xml --no-progress --stats
+
+      - name: Copy baseline
+        run: |
+          cp base/psalm.baseline.xml pr/psalm.baseline.xml
+
+      - name: Install dependencies for PR
+        run: |
+            cd pr
+            echo "::group::modify composer.json"
+            sed -i -re 's/"replace": \{/"replace": \{"symfony\/phpunit-bridge": "self.version",/' composer.json
+            composer require --no-update phpunit/phpunit php-http/discovery
+            echo "::endgroup::"
+            echo "::group::composer update"
+            composer update --no-progress --ansi
+            echo "::endgroup::"
+
+      - name: Cache Psalm
+        uses: actions/cache@v2
+        with:
+          path: pr/.github/.cache/psalm/
+          key: psalm-${{ github.sha }}
```

Comment:
github.base_ref instead? (same for the composer cache - of we could drop the composer cache btw)